libsmack/libsmack.c

```diff
@@ -576,14 +577,14 @@ const char *smack_smackfs_path(void)
 	return smackfs_mnt;
 }
 
-ssize_t smack_new_label_from_self(char **label)
+static ssize_t smack_new_label_from_proc(const char *proc_path, char **label)
 {
 	char buf[SMACK_LABEL_LEN + 1];
 	char *result;
 	int fd;
 	int ret;
```

Comment:
Thank you for review.

It might be good to fix this in entire source and use proper types. However in this particular case I don't think it would matter. `read()` is going to read at most `SMACK_LABEL_LEN` bytes so its return value will be in range <-1, 255>. I think that these values can be safely cast safely to `int` and back.

So I would ignore this one for now and fix type casting globally in a separate patch later.